Porting/add-pod-file

```diff
@@ -0,0 +1,242 @@
+         I               => 'Internals and C Language Interface',
+         M               => 'Miscellaneous',
+         L               => 'Language-Specific',
+         P               => 'Platform-Specific',
+
+For a first pass, we'll put the new entry at the end of the C<^=head2> section
+specified by the committer with the single-initial provided for command-line
+switch C<section>.
+
+=head2 Testing this program
+
+=over 4
+
+=item 1 Run F<configure> and F<make> in the source tree.
+
+=item 2 Create a well formatted F<.pod> file somewhere on your system.
+
+=item 3 Copy it into the source tree under F<pod>.
+
+=item 4 Call the program as in L</USAGE> above.
+
+=item 5 Call F<git diff> and examine results.
+
+=item 6 Run F<make test_porting>.
+
+=back
+
+=head1 BUGS
+
+When the argument provided to the C<--section> command-line switch is C<P> (for platform-specific), F<win32/pod.mak> is not getting updated -- but it's not clear whether it I<ought> to be updated.
+
+=cut
+
+my @man_sections = (
+    O   => 'Overview',
+    T   => 'Tutorials',
+    R   => 'Reference Manual',
+    I   => 'Internals and C Language Interface',
+    M   => 'Miscellaneous',
+    L   => 'Language-Specific',
+    P   => 'Platform-Specific',
+);
+
+my @man_section_abbrevs = ();
+my $man_sections_str = '';
+for (my $i=0; $i<= $#man_sections; $i+=2) {
+    my $j = $i+1;
+    push @man_section_abbrevs, $man_sections[$i];
+    $man_sections_str .= "\t$man_sections[$i] => $man_sections[$j]\n";
+}
+my %man_sections_seen = map { $_ => 1 } @man_section_abbrevs;
+my $man_sections = { @man_sections };
+
+my ($stub, $section, $verbose) = ('') x 3;
+GetOptions(
+    "stub=s"        => \$stub,
+    "section=s"     => \$section,
+    "verbose"       => \$verbose,
+) or croak("Error in command line arguments to add-pod-file.pl\n");
+croak "$0: Must provide value for command-line switch 'stub'"
+    unless length($stub);
+croak "$0: Must provide value for command-line switch 'section'"
+    unless length($section);
+my $section_croak = "$0: Value for command-line switch must be one of @man_section_abbrevs\n";
+$section_croak .= "    Select one initial from:\n$man_sections_str";
+croak $section_croak unless $man_sections_seen{$section};
+
+my $newpodfile = "$stub.pod";
+my $newpodpath = File::Spec->catfile('pod', $newpodfile);
+croak "Unable to locate new file '$newpodpath'" unless -f $newpodpath;
+my $thispodchecker = File::Spec->catfile(qw|cpan Pod-Checker podchecker|);
+croak "Cannot locate 'podchecker' within this checkout; have you called 'make'?"
+    unless -f $thispodchecker;
+
+say "Step 1: Basic test of validity of POD in $newpodpath" if $verbose;
+
+system(qq|$^X $thispodchecker $newpodpath|)
+    and croak "$newpodpath has POD errors; correct before proceeding further";
+my $data = Module::Metadata->new_from_file($newpodpath, collect_pod => 1, decode_pod => 1);
+
+my ($abstract) = ($data->pod('NAME') // '')  =~ / \A \s+ \S+ \s? - \s? (.+?) \s* \z /x
```

Comment:
This regex is too restrictive as I pointed out in my further comments

Reply:
I believe that for the purpose of this program, Leon's regex is sufficient.  It handles F<.pod> files with C<NAME> sections composed in the manner in which the overwhelming majority of existing F<pod/*.pod> files are composed.  If you can show me a potential new F<pod/*.pod> which needs your regex pattern, I will accept it.  Otherwise, I will not.

Thank you very much.
Jim Keenan

Reply:
Three examples:

Some modules like https://perldoc.perl.org/Encode::Supported have a double-dash before the abstract - this is arguably wrong but the dzil regex handles this.

Any documentation for a script containing a hyphen may be misparsed by that, since it doesn't require spaces before and after the hyphen.

Any documentation such as https://perldoc.perl.org/diagnostics with multiple program names will be misparsed.

Reply:
Additionally I will add - I would prefer the abstract parsing code to be more correct than less for the sake of future cargo culting :)